### ISSUE TITLE
[plugin-page-url-enrichment-browser] Previous Page incorrectly set to current page on direct visit in Next.js/SPAs

### ISSUE THREAD

**Issue by tmchow** [amplitude/Amplitude-TypeScript#1359]
## Expected Behavior

When a user directly visits a page (no referrer) for the first time in a session, the `@amplitude/plugin-page-url-enrichment-browser` plugin should set:
- `[Amplitude] Previous Page Location`: Empty string or the actual `document.referrer` value
- `[Amplitude] Previous Page Type`: `"direct"` (when no referrer exists)

The previous page should NOT be the same as the current page on a direct visit.

## Current Behavior

When a user directly visits a page in a Next.js application, the plugin incorrectly sets:
- `[Amplitude] Previous Page Location`: Same as current page URL
- `[Amplitude] Previous Page Type`: `"internal"`

This happens because the plugin's `saveURLInfo()` function is triggered during Next.js hydration/initialization (via `history.replaceState` proxy) before the first event is tracked. The storage gets initialized incorrectly:

1. Storage is empty initially
2. Next.js calls `history.replaceState()` during hydration (normal SPA behavior)
3. Plugin's proxy triggers `saveURLInfo()`
4. `saveURLInfo()` moves `AMP_CURRENT_PAGE` → `AMP_PREVIOUS_PAGE` (but current page is the only page!)
5. Session storage ends up with both current and previous set to the same URL

**Session Storage State After Bug:**
```json
{
  "AMP_CURRENT_PAGE": "https://www.example.com/",
  "AMP_PREVIOUS_PAGE": "https://www.example.com/"
}
```

## Possible Solution

The issue is in the [`saveURLInfo()` function](https://github.com/amplitude/Amplitude-TypeScript/blob/6cea10aa528d37483ca79759ab83d7a5c4178332/packages/plugin-page-url-enrichment-browser/src/page-url-enrichment.ts#L84-L95) in `page-url-enrichment.ts`:

```typescript
previousURL = (URLInfo?.[CURRENT_PAGE_STORAGE_KEY]) || '';
```

This blindly copies the current page to previous page on every navigation, even when it's the very first initialization.

**Suggested Fix:**

The plugin should distinguish between:
1. **First initialization** (no session storage exists yet) - Should use `document.referrer`
2. **Actual navigation** (storage exists, user is navigating) - Should move current → previous

Modified logic:
```typescript
const saveURLInfo = async () => {
  if (sessionStorage && isStorageEnabled) {
    const URLInfo = await sessionStorage.get(URL_INFO_STORAGE_KEY);
    const currentURL = getDecodeURI(location.href || '');

    // On first initialization, use document.referrer instead of current page
    const previousURL = URLInfo?.[CURRENT_PAGE_STORAGE_KEY]
      ? URLInfo[CURRENT_PAGE_STORAGE_KEY]  // Navigation: use previous current
      : (document.referrer || '');          // First load: use referrer

    await sessionStorage.set(URL_INFO_STORAGE_KEY, {
      [CURRENT_PAGE_STORAGE_KEY]: currentURL,
      [PREVIOUS_PAGE_STORAGE_KEY]: previousURL,
    });
  }
};
```

Alternatively, the `execute()` method already has logic to handle first initialization (lines 144-147), but `saveURLInfo()` gets called by the navigation proxies before `execute()` runs, causing the corruption.

## Steps to Reproduce

1. Install `@amplitude/plugin-page-url-enrichment-browser` in a Next.js application
2. Initialize Amplitude with the plugin:
```javascript
import * as amplitude from '@amplitude/analytics-browser';
import { pageUrlEnrichmentPlugin } from '@amplitude/plugin-page-url-enrichment-browser';

const pageUrlPlugin = pageUrlEnrichmentPlugin();
amplitude.add(pageUrlPlugin);
amplitude.init('YOUR_API_KEY');
```
3. Clear session storage (or open an incognito window)
4. Navigate directly to your homepage (e.g., type URL in address bar or bookmark)
5. Track a page view event after page load:
```javascript
amplitude.track('Page View');
```
6. Inspect the event properties

**Expected:**
- `[Amplitude] Previous Page Location`: `""` (empty)
- `[Amplitude] Previous Page Type`: `"direct"`

**Actual:**
- `[Amplitude] Previous Page Location`: `"https://www.example.com/"`
- `[Amplitude] Previous Page Type`: `"internal"`

**Session Storage Contents:**
```javascript
sessionStorage.getItem('AMP_URL_INFO')
// {"AMP_CURRENT_PAGE":"https://www.example.com/","AMP_PREVIOUS_PAGE":"https://www.example.com/"}
```

## Environment

- JS SDK Version: `@amplitude/analytics-browser@^2.16.1`, `@amplitude/plugin-page-url-enrichment-browser@^0.4.3`
- Installation Method: NPM
- Browser and Version: Chrome 131 (reproduced across browsers)
- Framework: Next.js 15.1.6

## Workaround

Pre-initialize the session storage before calling `amplitude.init()`:

```javascript
const urlInfoKey = 'AMP_URL_INFO';
const existingUrlInfo = window.sessionStorage.getItem(urlInfoKey);
if (!existingUrlInfo) {
  const initialUrlInfo = {
    AMP_CURRENT_PAGE: window.location.href,
    AMP_PREVIOUS_PAGE: document.referrer || '',
  };
  window.sessionStorage.setItem(urlInfoKey, JSON.stringify(initialUrlInfo));
}

amplitude.add(pageUrlPlugin);
amplitude.init('YOUR_API_KEY');
```

This prevents the plugin from incorrectly initializing the storage during Next.js hydration.
